pennylane/qnn/torch.py

```diff
@@ -0,0 +1,311 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains the classes and functions for integrating QNodes with the Torch Module
+API."""
+import functools
+import inspect
+import math
+from collections.abc import Iterable
+from typing import Callable, Optional
+
+try:
+    import torch
+    from torch.nn import Module
+    from pennylane.interfaces.torch import to_torch
+
+    TORCH_IMPORTED = True
+except ImportError:
+    # The following allows this module to be imported even if PyTorch is not installed. Users
+    # will instead see an ImportError when instantiating the TorchLayer.
+    from abc import ABC
+
+    Module = ABC
+    TORCH_IMPORTED = False
+
+
+class TorchLayer(Module):
+    r"""Converts a :func:`~.QNode` to a Torch layer.
+
+    The result can be used within the ``torch.nn``
+    `Sequential <https://pytorch.org/docs/stable/nn.html#sequential>`__ or
+    `Module <https://pytorch.org/docs/stable/nn.html#module>`__ classes for
+    creating quantum and hybrid models.
+
+    Args:
+        qnode (qml.QNode): the PennyLane QNode to be converted into a Torch layer
+        weight_shapes (dict[str, tuple]): a dictionary mapping from all weights used in the QNode to
+            their corresponding shapes
+        output_dim (int): the output dimension of the QNode
+        init_method (callable): a `torch.nn.init <https://pytorch.org/docs/stable/nn.init.html>`__
+            function for initializing the QNode weights. If not specified, weights are randomly
+            initialized using the uniform distribution over :math:`[0, 2 \pi]`.
+
```

Comment:
Thanks for checking the docs building @mariaschuld!

I just rebuilt and had a look and it looks ok:
![image](https://user-images.githubusercontent.com/49409390/81303238-6fbea980-9049-11ea-9d8f-fab9a66a3c2b.png)

Note that there were a few intermediate changes since the version you built, not sure if that might have inadvertently fixed things :thinking:

Reply:
Yes, it is working now :) But in the meantime I updated all my docs requirements, likely an old sphinx extension version or something...